From 624efa533536b243fd5aaed4853c129e9c1c9791 Mon Sep 17 00:00:00 2001
From: Neil Horman <nhorman@redhat.com>
Date: Wed, 22 Oct 2014 19:19:21 -0400
Subject: [scsi] fcoe: Fix missing mutex_unlock in fcoe_sysfs_fcf_add error path

Message-id: <1414005562-16279-8-git-send-email-nhorman@redhat.com>
Patchwork-id: 98087
O-Subject: [RHEL7.1 v2 PATCH 7/8] fcoe: Fix missing mutex_unlock in fcoe_sysfs_fcf_add error path
Bugzilla: 1087099
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>

In this pending patch:
http://patchwork.open-fcoe.org/patch/104/

Tomas Henzl noted that the error path when fcoe_fcf_device_add fails, was
missing a mutex_unlock call.

Not sure what staet the integration of the above patch is in, but if you could
either merge this with it, or apply it on top of what you already have, that
would be great.  Thanks!

Signed-off-by: Neil Horman <nhorman@tuxdriver.com>
CC: thenzl@redhat.com
Reported-by: thenzl@redhat.com
Signed-off-by: Robert Love <robert.w.love@intel.com>
(cherry picked from commit 55d0ac5d2839fe270cea02fad44eed13750a0efd)

BZ: 1087099
Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=8150961
Testing: Tested by Intel
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/fcoe/fcoe_ctlr.c b/drivers/scsi/fcoe/fcoe_ctlr.c
index 2c6b742..d25e2c3 100644
--- a/drivers/scsi/fcoe/fcoe_ctlr.c
+++ b/drivers/scsi/fcoe/fcoe_ctlr.c
@@ -203,6 +203,7 @@ static int fcoe_sysfs_fcf_add(struct fcoe_fcf *new)
   fcf_dev = fcoe_fcf_device_add(ctlr_dev, temp);
   if (unlikely(!fcf_dev)) {
    rc = -ENOMEM;
+   mutex_unlock(&ctlr_dev->lock);
    goto out;
   }
 
-- 
1.7.1